homeassistant/loader.py

```diff
@@ -590,11 +627,16 @@ def resolve_from_root(
                 )
                 continue
 
+            file_path = manifest_path.parent
+            # Avoid the listdir for virtual integrations
+            # as they cannot have any platforms
+            is_virtual = manifest.get("integration_type") == "virtual"
             integration = cls(
                 hass,
                 f"{root_module.__name__}.{domain}",
-                manifest_path.parent,
+                file_path,
                 manifest,
+                set() if is_virtual else set(os.listdir(file_path)),
```

Comment:
Since you're using a falsey check, an empty set will be replaced by an empty set in the constructor. 
```suggestion
                None if is_virtual else set(os.listdir(file_path)),
```